src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> previous;
+        private Node<T> next;
+
+        public Node(Node<T> previous, T value, Node<T> next) {
+            this.previous = previous;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void throwException(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index out of bounds for length");
+        }
+    }
+
+    private Node<T> getNodeByIndex(int targetIndex) {
+        throwException(targetIndex);
+        Node<T> currentNode = tail;
+        for (int i = size - 1; i != targetIndex; i--) {
```

Comment:
Optimize solution. When inserting node from head too.

Reply:
On my mind it isn't needed in my solution.